model/criteria.py

```diff
@@ -20,64 +26,154 @@ def check_criteria_against_paths(self, control_flow_graph, execution_paths):
+            to_return += 'Overall coverage : {:.2f}%'.format(
+                100 * len(self.covered) / len(self.to_cover)
+            )
+        except ZeroDivisionError:
+            to_return += 'Overall coverage : 0.00%'
+        for elt in self.to_cover:
+            if elt in self.covered:
+                to_return += '\nElement {} -- o'.format(elt)
+            else:
+                to_return += '\nElement {} -- x'.format(elt)
+        return to_return
+
 
 class TA(Criteria):
     """Get all the labels of the cover graph, and checks if they are all defined
     in the nodes of the programm"""
 
     def check_criteria_against_paths(self, control_flow_graph, execution_paths):
-        assigned = []
+        for vertex in control_flow_graph.vertices:
+            if vertex.operation == 'assignment':
+                self.to_cover.append(vertex)
+
         for path in execution_paths:
             for vertex in path:
                 if vertex.operation == 'assignment':
-                    assigned.append(vertex)
-
-        for vertex in control_flow_graph.vertices:
-            if vertex.operation == 'assignment' and vertex not in assigned:
-                return False
-        return True
+                    self.covered.append(vertex)
 
     def __repr__(self):
-        return "TA - All assignments"
+        return 'TA - All assignments'
 
 
 class TD(Criteria):
     """Get all the edges of type "while" or "if" and checks that they are evaluated"""
 
     def check_criteria_against_paths(self, control_flow_graph, execution_paths):
-        decided = []
+        for edge in control_flow_graph.edges:
+            if edge.root_vertex.operation in ['if', 'while']:
+                self.to_cover.append(edge.child_vertex)
+
         for path in execution_paths:
             for index, vertex in enumerate(path):
                 if vertex.operation in ['if', 'while']:
-                    decided.append(path[index + 1])
-
-        for edge in control_flow_graph.edges:
-            if edge.root_vertex.operation in ['if', 'while'] and edge.child_vertex not in decided:
-                return False
-        return True
+                    self.covered.append(path[index + 1])
 
     def __repr__(self):
-        return "TD - All decisions"
+        return 'TD - All decisions'
 
 
 class kTC(Criteria):
     """We get all the k paths from the cover graph (all the "small paths") and
     check if they are in the execution_paths we got from execution."""
 
     def __init__(self, k=1):
+        super().__init__()
         self.k = k
 
     def check_criteria_against_paths(self, control_flow_graph, execution_paths):
-        k_paths = []
+        # all the k paths from the cover graph
+        self.to_cover = control_flow_graph.get_all_k_paths(self.k)
+
         # we get all the possible paths
         for path in execution_paths:
-            k_paths.append(path[:self.k])
+            self.covered.append(path) if len(path) <= self.k else None
```

Comment:
If the execution path takes the same nodes than the k_path, but is longer, the k_path should be considered as covered right ? I think it's not the case here.